File: Cpp/201_Bitwise_AND_of_Numbers_Range.cpp
class Solution {
public:
    int rangeBitwiseAnd(int m, int n) {
        int begin = m, end = n, i = 0;
        while(begin != end){
            begin = begin >> 1;
            end = end >> 1;
            ++i;
        }
        return begin << i;
    }
};

File: Kotlin/201_Bitwise_AND_of_Numbers_Range.kt
class Solution {
    fun rangeBitwiseAnd(left: Int, right: Int): Int {
        var begin = left
        var end = right
        var i = 0
        
        while(begin != end) {
            begin = begin shr 1
            end = end shr 1
            ++i
        }
        
        return begin shl i
    }
}